python/GafferDispatchUI/DispatchDialogue.py

```diff
@@ -0,0 +1,261 @@
+import IECore
+
+import Gaffer
+import GafferDispatch
+
+import GafferUI
+
+## A dialogue which can be used to dispatch tasks
+class DispatchDialogue( GafferUI.Dialogue ) :
+
+	def __init__( self, script, tasks, dispatcherType, applyUserDefaults=False, title="Dispatch Tasks", sizeMode=GafferUI.Window.SizeMode.Manual, **kw ) :
+
+		GafferUI.Dialogue.__init__( self, title, sizeMode=sizeMode, **kw )
+
+		self._getWidget().setBorderStyle( GafferUI.Frame.BorderStyle.None )
+
+		# build tabs for all the node, dispatcher, and context settings
+		with GafferUI.ListContainer() as self.__settings :
+			with GafferUI.TabbedContainer() as self.__tabs :
+
+				with GafferUI.ListContainer( spacing=2, borderWidth=2 ) as dispatcherTab :
+					with GafferUI.ListContainer( GafferUI.ListContainer.Orientation.Horizontal, spacing=2, borderWidth=2 ) :
+						GafferUI.Label( "Dispatcher" )
+						self.__dispatchersMenu = GafferUI.MultiSelectionMenu( allowMultipleSelection = False, allowEmptySelection = False )
+						self.__dispatchersMenu.append( list(GafferDispatch.Dispatcher.registeredDispatchers()) )
+						self.__dispatchersMenu.setSelection( [ dispatcherType ] )
+						self.__dispatchersMenuChanged = self.__dispatchersMenu.selectionChangedSignal().connect( Gaffer.WeakMethod( self.__dispatcherChanged ) )
+
+					self.__dispatcherFrame = GafferUI.Frame( borderStyle=GafferUI.Frame.BorderStyle.None, borderWidth=0 )
+					self.__tabs.setLabel( dispatcherTab, "Dispatcher" )
+
+				with GafferUI.Frame( borderStyle=GafferUI.Frame.BorderStyle.None, borderWidth=2 ) as contextTab :
+					GafferUI.PlugValueWidget.create( script["variables"] )
+					self.__tabs.setLabel( contextTab, "Context Variables" )
+
+		# build a ui element for progress feedback and messages
+		with GafferUI.ListContainer( spacing = 4 ) as self.__progressUI :
+
+			with GafferUI.ListContainer( parenting = { "horizontalAlignment" : GafferUI.HorizontalAlignment.Center, "verticalAlignment" : GafferUI.VerticalAlignment.Center } ) :
+				self.__progressIconFrame = GafferUI.Frame( borderStyle = GafferUI.Frame.BorderStyle.None, parenting = { "horizontalAlignment" : GafferUI.HorizontalAlignment.Center } )
+				self.__progressLabel = GafferUI.Label( parenting = { "horizontalAlignment" : GafferUI.HorizontalAlignment.Center } )
+
+			with GafferUI.Collapsible( "Details", collapsed = True, parenting = { "expand" : True } ) as self.__messageCollapsible :
+				self.__messageWidget = GafferUI.MessageWidget()
+				# connect to the collapsible state change so we can increase the window
+				# size when the details pane is first shown.
+				self.__messageCollapsibleConneciton = self.__messageCollapsible.stateChangedSignal().connect( Gaffer.WeakMethod( self.__messageCollapsibleChanged ) )
+
+		self.__button = self._addButton( "Dispatch" )
+
+		self.__script = script
+		# hold a reference to the script window so plugs which launch child windows work properly
+		self.__scriptWindow = GafferUI.ScriptWindow.acquire( script )
```

Comment:
I also don't like needing this one, but it seemed the easiest way to ensure color swatches and ramps worked as expected.

Reply:
Addressed in 57a006132da5bfa4a0ccfce3d710a6a17249a467